Docs site audit part 2 - inline link cleanup

validbeck opened the pull request:
## External Release Notes

> For sc-4898, I:
>
>- Added `vm_login` to the `_variables.yml` file & changed all instances of "1. [Log in to the {{< var vm_platform >}}]({{< var vm_url >}})." to "1. Log in to the {{< var vm_login >}}."
>- Adjusted the hacky CSS on `customize-documentation-templates.qmd` to match the new link styling applied in PR#217
>- Established a convention in the style guide for `.column-margin` content and used this to pull out extraneous links from step-by-step instructions or text
>- Reviewed extraneous bolding of links where appropriate
>- Retooled the "About ValidMind" landing page to include "Automated testing & documentation" & "Model risk governance management" (removed the "Features" section from About)

### Major changes

#### variables.yml
> [Added `vm_login`:](https://github.com/validmind/documentation/blob/c55094aa219ee51e865eb572190e9ce6a656f0a7/site/_variables.yml#L14) `"[ValidMind Platform UI](https://app.prod.validmind.ai)"`
- Changed all instances of "1. [Log in to the {{< var vm_platform >}}]({{< var vm_url >}})." to "1. Log in to the {{< var vm_login >}}."
- For some articles that did not have the first login step but required it, I made a quick adjustment as I was working

#### Customize documentation templates
> Adjusted the [hacky CSS on `customize-documentation-templates.qmd`](https://github.com/validmind/documentation/blob/c55094aa219ee51e865eb572190e9ce6a656f0a7/site/guide/customize-documentation-templates.qmd#L56) to match the new link styling applied in PR#217

#### .column-margin content
> New section in the style guide for inline links/margin content rationale & guidance:
<img width="1053" alt="style-guide_margins" src="https://github.com/validmind/documentation/assets/164545837/9049918d-3ccf-4f04-b3f6-31383abab21b">
- Reviewed all articles for extraneous links, especially in step-by-step instructions and applied this new margin styling instead when appropriate (or did not mess with single-sourced training, etc.):

| Old | New |
|---|---|
| <img width="976" alt="inline-links_incorrect" src="https://github.com/validmind/documentation/assets/164545837/f37e9304-187e-4e90-86e5-ecc59bc810fb"> | <img width="1025" alt="margin-content_correct" src="https://github.com/validmind/documentation/assets/164545837/18feb30a-7453-4017-84c2-3dc556e6c400">|

#### About 
> Retooled the "About ValidMind" landing page to include "Automated testing & documentation" & "Model risk governance management" (removed the "Features" section from About)
- Now there are some fancy cards under "ValidMind AI risk platform" with links out to the relevant docs instead of the nebulous `Read more ...`
- The two linked pages got some quick cleanup as well to remove the `Read more ...`s 

| Page | Old | New |
|---|---|---|
| About ValidMind | <img width="1397" alt="about-old" src="https://github.com/validmind/documentation/assets/164545837/2a3b2022-952b-4eb5-8f80-8709ac15dd60"> | <img width="1378" alt="Screenshot 2024-06-20 at 1 59 05 PM" src="https://github.com/validmind/documentation/assets/164545837/81a79c9b-466f-48e5-b5e0-0a700697ca48"> |
| Automated testing & documentation | <img width="986" alt="old-automation" src="https://github.com/validmind/documentation/assets/164545837/611784a6-b8bf-42e7-8d4e-c4f9318f61a8"> | <img width="1045" alt="new-automation" src="https://github.com/validmind/documentation/assets/164545837/db2f835b-a7d0-401f-a691-487d101ecf46"> |
| Model risk governance management | <img width="1020" alt="new-mrm" src="https://github.com/validmind/documentation/assets/164545837/7eafb3f5-8fa5-44e6-9338-7650734517f3"> | <img width="975" alt="old-mrm" src="https://github.com/validmind/documentation/assets/164545837/1398754b-5117-43c5-9fa8-829d5b0453b4"> |

### Other changes
> Following our own rules about minimizing callouts and unnecessary repeats of text and tricky conditional displays etc.

| Page | Old | New |
|---|---|---|
| Review model documentation | <img width="995" alt="review-docs_old" src="https://github.com/validmind/documentation/assets/164545837/223ffdf4-b4d6-49fc-919b-f50c464aa25a"> | <img width="753" alt="review-docs_new" src="https://github.com/validmind/documentation/assets/164545837/774f7831-543b-4a06-a671-77ffa1bde128"> |
| Collaborate with others | <img width="1053" alt="collab_old" src="https://github.com/validmind/documentation/assets/164545837/09e30a4c-e4b8-45d1-9dae-593a34c39e76"> | <img width="741" alt="collab_new" src="https://github.com/validmind/documentation/assets/164545837/6aa50f03-3bfc-4445-aa10-fc2a1b8550ca"> |

### Other notes
> I noticed that some images don't have alt text as per our style guide so I will add that to the list for the next review. 
- I'm going to wait until Noosheen's PR #219 is merged into `main` so I can pull in those changes and do a final pass before merging this PR. **EDIT:** DONE :) Thanks Noosheen!

nrichers reviewed (commented):
LGTM! My only comment is that we should use bullets OR superscript numbers but not both. I'd globally change callouts to look like this: 

<img width="967" alt="image" src="https://github.com/validmind/documentation/assets/15148011/46670bed-734c-4641-8b10-fb86aa4e34bc">

validbeck commented:
@nrichers:  

Just removed the numeration from this page:
<img width="990" alt="Screenshot 2024-06-20 at 4 38 00 PM" src="https://github.com/validmind/documentation/assets/164545837/cb7b3784-9fa2-4d02-9c28-daa82ae2e0d5">

Removed the bullet points from any pages with margin lists:
<img width="995" alt="Screenshot 2024-06-20 at 4 57 27 PM" src="https://github.com/validmind/documentation/assets/164545837/e823a0d1-21e1-47fa-835c-bf1c478d5326">
